doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md

```diff
@@ -0,0 +1,359 @@
+     */
+    SAI_ROUTER_INTERFACE_ATTR_LOOPBACK_PACKET_ACTION,
+
+} sai_router_interface_attr_t;
+```
+
+## 2.5 Statistics
+Packets that are dropped due to loopback action will be counted in TX_ERR in IP interface statistics.
+```
+root@r-lionfish-14:/home/admin# show interfaces counters rif
+       IFACE    RX_OK    RX_BPS    RX_PPS    RX_ERR    TX_OK    TX_BPS    TX_PPS    TX_ERR
+------------  -------  --------  --------  --------  -------  --------  --------  --------
+ Ethernet236        4  0.00 B/s    0.00/s         1        0  0.00 B/s    0.00/s         0
+PortChannel1        0  0.00 B/s    0.00/s         0        0  0.00 B/s    0.00/s         0
+       Vlan2        0  0.00 B/s    0.00/s         0        0  0.00 B/s    0.00/s         0
+Ethernet0.10        0  0.00 B/s    0.00/s         0        0  0.00 B/s    0.00/s         0
+```
+
+## 2.6 CLI
+### 2.6.1 Config command
+A new SONiC CLI command is introduced to configure IP interface loopback action.
+```
+# Set loopback action per INTERFACE/PORTCHANNEL_INTERFACE/VLAN_INTERFACE/VLAN_SUB_INTERFACE
+config interface loopback-action <intf-name> drop|forward
+```
+
+Examples:
+```
+config interface loopback-action Ethernet248 drop
+config interface loopback-action Vlan100 forward
+```
+
+### 2.6.2 Show command
+A new SONiC CLI command is introduced to show the configured loopback action.
+IP interfaces that user did not configure loopback action on them, will not be shown in the table.
+```
+show ip interfaces loopback-action
+```
+
+Example:
+```
+root@sonic:~# show ip interfaces loopback-action
+Interface     Action      
+------------  ----------  
+Ethernet232   drop
+Vlan100       forward     
+```
+
+## 2.7 YANG model
+The following YANG models will be extended in order to provide support for loopback action:
+1. sonic-interface.yang -> container INTERFACE 
+2. sonic-portchannel.yang -> container PORTCHANNEL_INTERFACE
+3. sonic-vlan.yang -> container VLAN_INTERFACE 
+4. sonic-vlan-sub-interface.yang -> container VLAN_SUB_INTERFACE
+
+New leaf loopback_action will be added:
+```
+leaf loopback_action {
+    description "Packet action when a packet ingress and gets routed on the same IP interface";
+    type string {
+        pattern "drop|forward";
+    }
+}
+```
+
+## 2.8 Warm/Fast boot
+No special handling is required
+
+# 3 Test plan
+## 3.1 Unit tests via VS
+Add the following test case in file sonic-swss/tests/test_interface.py.
+1. Create IP interface by writing to config DB
+2. Set loopback action on IP interface
+3. Check table INTF_TABLE in App DB
+4. Check table SAI_OBJECT_TYPE_ROUTER_INTERFACE in ASIC DB
+5. Create port-channel IP interface and repeat steps 2 to 4.
+6. Create interface VLAN and repeat steps 2 to 4.
+7. Create subinterface and repeat steps 2 to 4.
+
+## 3.2 System tests
+Add the following test in sonic-mgmt.
+1. Create IP interface Ethernet.
+2. Loopback action not configured, send traffic, verify lopbacked packets are forwarded (SAI default).
```

Comment:
loopback -> typo. Please fix other places as well

Reply:
Fixed in all places.